test/extended/router/gatewayapi.go

```diff
@@ -0,0 +1,209 @@
+	defer g.GinkgoRecover()
+	var (
+		oc       = exutil.NewCLIWithPodSecurityLevel("gateway-api", admissionapi.LevelBaseline)
+		crdNames = []string{
+			"gatewayclasses.gateway.networking.k8s.io",
+			"gateways.gateway.networking.k8s.io",
+			"httproutes.gateway.networking.k8s.io",
+			"referencegrants.gateway.networking.k8s.io",
+		}
+		errorMessage = "ValidatingAdmissionPolicy 'openshift-ingress-operator-gatewayapi-crd-admission' with binding 'openshift-ingress-operator-gatewayapi-crd-admission' denied request: Gateway API Custom Resource Definitions are managed by the Ingress Operator and may not be modified"
+	)
+
+	g.Describe("The Gateway API CRDs", func() {
+		g.It("should already be installed", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Get and check the installed CRDs")
+			for i := range crdNames {
+				crd, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), crdNames[i], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				e2e.Logf("Found the Gateway API CRD named: %v", crd.Name)
+			}
+		})
+
+		g.It("should not be deleted", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Trying to delete the CRDs")
+			for i := range crdNames {
+				err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Delete(context.Background(), crdNames[i], metav1.DeleteOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+			}
+		})
+
+		g.It("should not be updated", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			g.By("Get the CRDs firstly, add spec.names.shortNames then update CRD")
+			for i := range crdNames {
+				crd, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Get(context.Background(), crdNames[i], metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				// some CRDs have a shortName but some not, just trying to add one for all
+				crd.Spec.Names.ShortNames = append(crd.Spec.Names.ShortNames, "fakename")
+				_, err = crdClient.ApiextensionsV1().CustomResourceDefinitions().Update(context.Background(), crd, metav1.UpdateOptions{})
+				o.Expect(err).To(o.HaveOccurred())
+				o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+			}
+		})
+
+		g.It("should not be created", func() {
+			crdClient := apiextensionsclientset.NewForConfigOrDie(oc.AdminConfig())
+			fakeCRDName := "fakeroutes.gateway.networking.k8s.io"
+			g.By("Creating new CRD in the group")
+			fakeCRD := buildGWAPICRDFromName(fakeCRDName)
+			_, err := crdClient.ApiextensionsV1().CustomResourceDefinitions().Create(context.Background(), fakeCRD, metav1.CreateOptions{})
+			o.Expect(err).To(o.HaveOccurred())
+			o.Expect(err.Error()).To(o.ContainSubstring(errorMessage))
+		})
+	})
+})
+
+// Adding [Serial] since it could break Gateway API CRDs creation/deletion/update tests
```

Comment:
You can create a fake CRD from the gateway API group and try to update and remove it. This would minimize any conflicts with existing CRDs.

Reply:
As a matter of fact, you already do this in `The Gateway API CRDs should not be created`. You can use the same fake crd for all the tests, why trying to delete real CRDs even if you expect them to fail due to the VAP protection?

Reply:
Wait, I see now why you used the real CRDs - to check that CIO recreates them. But isn't this already tested in `The Gateway API CRDs should already be installed` (CIO manage to create them when it first started)?

Reply:
Adding [Serial] since the last test case needs scale down CVO and removes the VAP, I cannot run in parallel with other 4 cases. The case is to test the "unsupported" config path that you mentioned in the slack.

Reply:
I updated the comment and hope that explains it more clearly.

Reply:
I removed this case and added new one of verifying experimental group CRDs. All five tests could be run in parallel now. Thanks.